javalib/src/main/scala/java/util/AbstractList.scala

```diff
@@ -253,11 +258,14 @@ private class RandomAccessListIterator[E](list: List[E], private var i: Int,
 
   def add(e: E): Unit = {
     list.add(i, e)
-    end += 1
+    onSizeChanged(1)
     last = -1
     i += 1
   }
 
+  protected def onSizeChanged(delta: Int): Unit =
+    end += delta
```

Comment:
Then, I think the actual change of `end` should not be part of `onSizeChanged`. `end` should first be updated, then `onSizeChanged` should be called. It seems a bit weird that the "default" implementation of an event handler would actually *do* things. Maybe my point of view is still influenced by Delphi's VCL design, though.